HeterogeneousCore/AlpakaCore/interface/alpaka/EventSetup.h

```diff
@@ -31,7 +32,11 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::device {
     template <typename T, typename R>
     T const& getData(device::ESGetToken<T, R> const& iToken) const {
       auto const& product = setup_.getData(iToken.underlyingToken());
-      return product.get(device_);
+      if constexpr (std::is_same_v<typename detail::ESDeviceProductType<T>::type, T>) {
```

Comment:
This pattern `std::is_same_v<typename detail::ESDeviceProductType<T>::type, T>` seems to be used frequently.

Maybe it could be wrapped as
```c++
template <typename T>
inline constexpr bool isDeviceESProductType = std::is_same_v<typename detail::ESDeviceProductType<T>::type, T>;
```
?

Reply:
or something more explicit than "not the same as" ?

Reply:
Yes, it can be simplified, thanks for the suggestion.

Reply:
I added `useESProductDirectly` variable, and made similar changes in the Event side as well with `useProductDirectly`